src/Rules/Username.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ARKEcosystem\Fortify\Rules;
+
+use ARKEcosystem\Fortify\Support\Enums\Constants;
+use Illuminate\Support\Str;
+use Laravel\Fortify\Rules\Password as Fortify;
+
+class Username extends Fortify
+{
+    /**
+     * Indicates if the username contains some forbidden special characters.
+     *
+     * @var bool
+     */
+    protected $withForbiddenSpecialChars = false;
+
+    /**
+     * Indicates if the username contains a special character at the start of it.
+     *
+     * @var bool
+     */
+    protected $withSpecialCharAtTheStart = false;
+
+    /**
+     * Indicates if the username contains a special character at the end of it.
+     *
+     * @var bool
+     */
+    protected $withSpecialCharAtTheEnd = false;
+
+    /**
+     * Indicates if the username contains consecutive special characters.
+     *
+     * @var bool
+     */
+    protected $withConsecutiveSpecialChars = false;
+
+    /**
+     * Determine if the validation rule passes.
+     *
+     * @param string $attribute
+     * @param mixed  $value
+     *
+     * @return bool
+     */
+    public function passes($attribute, $value)
+    {
+        // Handle potential NULL values
+        $value = $value ?: '';
+
+        if ($this->withForbiddenSpecialChars($value)) {
+            $this->withForbiddenSpecialChars = true;
+
+            return false;
+        }
+
+        if ($this->withSpecialCharAtTheStart($value)) {
+            $this->withSpecialCharAtTheStart = true;
+
+            return false;
+        }
+
+        if ($this->withSpecialCharAtTheEnd($value)) {
+            $this->withSpecialCharAtTheEnd = true;
+
+            return false;
+        }
+
+        if ($this->withConsecutiveSpecialChars($value)) {
+            $this->withConsecutiveSpecialChars = true;
+
+            return false;
+        }
+
+        return ! $this->needsMinimumLength($value);
+    }
+
+    /**
+     * Get the validation error message.
+     *
+     * @return string
+     */
+    public function message()
+    {
+        if ($this->message) {
+            return $this->message;
+        }
```

Comment:
in what case does a message get set?

Reply:
Atm, none, but just left the possibility like for the password one, cause I don't think we set different ones there either, but since the option was available 🤷 But can be removed yeah